### PR TITLE
add contract examples

### PR DESCRIPTION
**Pull request by ceyonur**
Creates contract examples for AllowList & NativeMinter precompiles

### CODE REVIEW

File: contract-examples/.gitignore [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/dist
```
Comment (aaronbuchwald) [inline contract-examples/.gitignore:1]: Should we move this to the repo's own gitignore or keep in this subdirectory?

File: contract-examples/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+```zsh
+$ git clone https://github.com/ava-labs/subnet-evm.git
+$ cd smart-contract-examples
+$ yarn
+```
+
+## Write Contracts
+
+`ERC20NativeMinter.sol` is an [Open Zeppelin](https://openzeppelin.com) [ERC20](https://eips.ethereum.org/EIPS/eip-20) contract powered by native minting capabilities of Subnet EVM. ERC20 is a popular smart contract interface. It uses `INativeMinter` interface to interact with `NativeMinter` precompile.
```
Comment (aaronbuchwald) [inline contract-examples/README.md:39]: nit: it seems like this says this is an OpenZeppelin contract. Can we clarify that it is based on an OpenZeppelin contract with added capabilities?

File: contract-examples/contracts/ERC20NativeMinter.sol [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  string private TOKEN_NAME = "ERC20NativeMinterToken";
+  string private TOKEN_SYMBOL = "XMPL";
+
+  event Deposit(address indexed dst, uint256 wad);
+  event Mintdrawal(address indexed src, uint256 wad);
+
+  constructor(uint256 initSupply) ERC20(TOKEN_NAME, TOKEN_SYMBOL) {
+    // Mints INIT_SUPPLY to owner
+    _mint(_msgSender(), initSupply);
+  }
```
[the remark this answers is not in the document]
Comment (ceyonur) [reply contract-examples/contracts/ERC20NativeMinter.sol:23]: It's 18 by default: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/52eeebecda140ebaf4ec8752ed119d8288287fac/contracts/token/ERC20/ERC20.sol#L87-L89

File: contract-examples/.gitignore (hunk 1 above)
Comment (ceyonur) [reply contract-examples/.gitignore:1]: I think it works within directories. and it's probably best to treat this one as a different folder (other than go packages)

File: contract-examples/contracts/ERC20NativeMinter.sol (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrick-ogrady) [reply contract-examples/contracts/ERC20NativeMinter.sol:23]: I think it may be worth clarifying

[the remark this answers is not in the document]
Comment (ceyonur) [reply contract-examples/contracts/ERC20NativeMinter.sol:24]: done

File: contract-examples/README.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
+# Subnet EVM Contracts
+
```
Comment (patrick-ogrady) [inline contract-examples/README.md:2]: Add in bold that contracts are not yet audited

File: contract-examples/README.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
+# Subnet EVM Contracts
```
Comment (patrick-ogrady) [inline contract-examples/README.md:1]: You should link to here from main README

File: contract-examples/README.md [hunk 6]
```diff
@@ -1,3 +1,5 @@
+CONTRACTS HERE ARE CONSIDERED IN ALPHA PHASE. USE AT YOUR OWN RISK! AVA LABS TAKES NO RESPONSIBILITY OR LIABILITY IF THERE IS A BUG IN IMPLEMENTATIONS.
```
Comment (patrick-ogrady) [inline contract-examples/README.md:1]: You can put this right under `Subnet EVM Contracts`

**Review by patrick-ogrady — APPROVED** [review]

**Review by aaronbuchwald — APPROVED** [review]
LGTM

**Pull request merged**